CONTRIBUTING.md

```diff
@@ -81,39 +81,55 @@ cherry-picked into `book-v2` branch.
+## Introducing changes to plugins
 
-**More simply:** any change that will break the expected behaviour of a project built with the previous `MINOR` Plugin versions is a breaking change to that plugin. 
+Changes made to plugins may require a plugin version increase depending on whether
+they break the current version or not. See the plugins design doc [versioning section][plugins-design-versioning]
+for more details.
 
-**EXAMPLE:**
+Plugin changes only warrant a new KubeBuilder release if and only if a plugin
+version is no longer supported by the CLI.
 
-KubeBuilder Release version (`5.3.1`) scaffolds projects with the plugin version `3.2.1` by default.
+### Example
 
-The changes introduced in our PR will not work well with the projects which were built with the plugin versions `3.0.0...3.5.1` without users taking manual steps to update their projects. Thus, our changes introduce a breaking change to the Go plugin, and require a `MAJOR` Plugin version bump.
+KubeBuilder release version `v2.3.1` scaffolds projects with plugin `go.kubebuilder.io/v2.1`
+by default. Previously, plugin `go.kubebuilder.io/v2.0` was used.
 
-In the PR, we should add a migration guide to the [Migrations](https://book.kubebuilder.io/migrations.html) section of the KubeBuilder book. It should detail the required steps that users should take to upgrade their projects from `go.kubebuilder.io/3.X.X` to the new `MAJOR` Plugin version `go.kubebuilder.io/4.0.0`.
+You make a PR that adds a new marker to the file `main.go` scaffolded by `init`
+that `create api` will use to update that file. The changes introduced in your PR
+will cause errors when used with projects built with plugins `go.kubebuilder.io/v2.0`
+and `go.kubebuilder.io/v2.1` without users manually updating their projects.
+Thus, your changes introduce a breaking change to plugin `go.kubebuilder.io`,
+and require a _major_ version increase to `v3.0` in the PR.
 
-This also means we should introduce a new KubeBuilder minor version `5.4.0` when the project is released: since we've only added a new plugin version without removing the old one, this is considered a new feature to KubeBuilder, and not a breaking change.
+You must also add a migration guide to the [migrations](https://book.kubebuilder.io/migrations.html)
+section of the KubeBuilder book in your PR. It should detail the steps required
+for users to upgrade their projects from `go.kubebuilder.io/v2.Y` to the new
+plugin version `v3.0`.
 
-**IMPORTANT** Breaking changes cannot be made to PROJECT versions v1 and v2, and consequently plugin versions `go.kubebuilder.io/1` and `go.kubebuilder.io/2`.
+You've only added a new plugin version without removing old versions, so this
+considered a new feature to KubeBuilder and not a breaking change. Therefore
+a new _minor_ KubeBuilder version `v2.4.0` should be release.
+
```

Comment:
The explanation here lead us to believe that after each change we need to do a release, which shows not true at all.
```suggestion
considered a new feature to KubeBuilder and not a breaking change. Therefore
if we decide to release a new version of KubeBuilder then, a new _minor_ KubeBuilder version with the tag `v5.4.0` should be done.
```